### PR TITLE
fix(web/mailer-settings): accept display-name form in From address field

### LINKED ISSUE #1749: [BUG] Frontend — mailer From address field rejects display-name format
## Problem / Context

The "Edit mailer settings" dialog rejects a From address that carries a display name (e.g. `OpenLinker <noreply@openlinker.io>`), showing **"Enter a valid email address"**. Admins can therefore only set a bare address, so recipients see `noreply@openlinker.io` in their inbox instead of a friendly sender name like **OpenLinker**.

The block is **frontend-only**. The backend already accepts the `Display Name <email>` form end-to-end:

- Request DTO `apps/api/src/mailer/http/dto/update-mailer-settings.dto.ts:40-47` — `fromAddress` is only `@IsOptional() @IsString()`, no `@IsEmail()`.
- Core service `libs/core/src/mailer/application/services/mailer-settings.service.ts` — `updateSettings()` / `resolveFromRow()` pass `fromAddress` through unchanged, no format check.
- Send path `apps/api/src/auth/adapters/smtp-mailer.adapter.ts:46-53` passes `from: this.from` straight into nodemailer's `sendMail`, and nodemailer natively parses the `"Display Name <email@domain>"` format into a proper `From:` header.

The only rejecting validator is the Zod schema regex on the frontend.

## Proposed Solution

Loosen the From-address validation in `apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts` so it accepts **both** a bare email and the `Display Name <email>` form.

- `EMAIL_PATTERN` (line 22) currently is `/^[^\s@]+@[^\s@]+\.[^\s@]+$/`, applied in the `superRefine` at lines 75-81 (only when `transport === 'smtp'`).
- Accept either:
  - a bare address matching the existing pattern, **or**
  - a `Name <address>` form where the part inside `<...>` matches the existing email pattern (validate the address inside the angle brackets; allow a non-empty display-name prefix).
- Keep the existing `MAX_FROM_ADDRESS_LENGTH` (320) cap and the "required for SMTP transport" check unchanged.
- Update the error message so it reflects both accepted forms (e.g. `Enter a valid email address, optionally with a display name (e.g. "OpenLinker <noreply@example.com>")`).

No backend, DTO, or core change is needed.

## Classification

**Type**: Frontend
**Layer**: Interface (form validation)
**File(s)**:
- `apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts` (fix)
- `apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.test.ts` (add/update tests if present; else colocate a new `*.test.ts`)

## Dependencies

- None.

## Assumptions

- Display-name form is only needed for SMTP transport (matches the existing `superRefine` gate). Console transport is unaffected.
- We validate only structural shape client-side; the backend remains the source of truth (per FE architecture) and nodemailer handles the actual header construction.

## Acceptance Criteria

- [ ] Entering `OpenLinker <noreply@openlinker.io>` in the From address field passes validation and saves successfully.
- [ ] A bare address (`noreply@openlinker.io`) still passes validation.
- [ ] An invalid value (e.g. `not-an-email`, `Name <not-an-email>`, empty for SMTP) still fails with a clear message.
- [ ] The 320-char max-length cap and the "required for SMTP transport" behavior are preserved.
- [ ] After saving `OpenLinker <noreply@openlinker.io>`, outbound mail shows `OpenLinker` as the sender display name (nodemailer builds the `From:` header from the stored value).
- [ ] Unit tests cover: bare email accepted, `Name <email>` accepted, invalid rejected, empty-for-SMTP rejected.
- [ ] No architecture boundary violations (frontend-only change).

### PR DESCRIPTION
**Pull request by jakubretajczykBD**
## Summary

- Loosens the mailer settings "From address" Zod validation to accept both a bare email (`noreply@openlinker.io`) and the `Display Name <email>` form (`OpenLinker <noreply@openlinker.io>`), matching what the backend already accepts end-to-end (DTO has no `@IsEmail()`, core service passes the value through unchanged, nodemailer parses this format natively into the `From:` header).
- Adds a helper hint under the field (`Also accepts a display name, e.g. "OpenLinker <orders@example.com>".`) so admins discover the syntax without hitting a validation error first, and switches the input off `type="email"` now that it intentionally accepts non-email syntax.
- Updates the validation error message to reflect both accepted forms.
- Hardens the regex against malformed input that would otherwise slip through or be misparsed:
  - a second bracketed address (`A <b@c.com> <d@e.com>`) is rejected instead of silently picking the wrong one
  - a stray angle bracket swallowed into the local-part (`Test <,<test@test.test>`) is rejected
  - consecutive dots / a leading dot in the local-part or domain (`test..test@test.pl`) are rejected
- Frontend-only change — no backend, DTO, or core changes needed.

## Related issues

Closes #1749

## Test plan

- [x] `noreply@openlinker.io` (bare email) passes validation
- [x] `OpenLinker <noreply@openlinker.io>` (display-name form) passes validation and saves successfully — verified manually via the Edit mailer settings dialog and confirmed in the `mailer_settings` table
- [x] Invalid values (`not-an-email`, `Name <not-an-email>`, empty for SMTP) still fail with a clear message
- [x] Malformed edge cases rejected: `A <b@c.com> <d@e.com>` (ambiguous double bracket), `Test <,<test@test.test>` (stray bracket in local-part), `test..test@test.pl` (consecutive dots)
- [x] 320-char max-length cap and "required for SMTP transport" behavior preserved
- [x] New unit tests added in `mailer-settings-form.schema.test.ts` covering all the above (10 cases total)
- [x] `pnpm test` passes (full web suite)

## Quality gate

- [x] `pnpm lint` passes (zero errors)
- [x] `pnpm type-check` passes (zero errors)
- [x] `pnpm test` passes (all unit tests green)
- [x] *Optional, Docker required:* `pnpm test:integration` passes — needed
      only if you touched `apps/api/test/integration/**` or any plugin's
      `infrastructure/adapters/`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<sub>By submitting this pull request, I confirm that my contributions
are made under the terms of the [Apache License 2.0](../LICENSE), and I
certify the [Developer Certificate of Origin](https://developercertificate.org/)
by signing off my commits.</sub>

### CODE REVIEW

**Comment by norbert-kulus-blockydevs** [comment]
## Review

**Overall: good to merge, with one thing worth tightening before merge (non-blocking) and a few minor follow-ups.**

Small, well-scoped change: it loosens the Zod validation on the mailer settings "From address" field to accept `Display Name <email>` in addition to a bare email. I independently verified the backend claim rather than taking the PR description at face value: the DTO (`update-mailer-settings.dto.ts`) only has `@IsString()`, the core service and repository pass the value through unchanged, the DB column is `text` with no length constraint, and `SmtpMailerAdapter` hands the string straight to nodemailer, which natively parses this RFC 5322 form. The frontend's 320-char cap is stricter than the backend, so there's no FE/BE drift risk. This field only backs auth-lifecycle mail (email confirmation, password reset) - it's a global singleton, not per-tenant - so there are no multi-tenant concerns, and it has no overlap with invoicing (KSeF/inFakt), which doesn't use this mailer at all.

### Important
- **No affordance in the field itself.** `mailer-settings-dialog.tsx:216` still has `placeholder="orders@example.com"` and `type="email"`. The new, longer error message explains the format, but only *after* a failed attempt. An admin who wants to set a friendly sender name has no hint that it's even possible until they hit a validation error with the wrong syntax. Worth adding a short helper text under the field since we're already touching it.
- **The regex has a documented-but-untested gap.** `NAME_AND_EMAIL_PATTERN` is greedy (`.+\s<...>$`), so `"A <b@c.com> <d@e.com>"` validates successfully but effectively silently picks `d@e.com` and discards the first address. The implementation plan committed alongside this PR explicitly calls out this exact case as a risk and recommends adding a test for it, but no such test made it into `mailer-settings-form.schema.test.ts` - that's a real drift between the plan and what shipped. Low odds a real user hits this, but since it's a named, known gap it should either get a regression test documenting the accepted behavior, or a small regex fix (excluding `<`/`>` from the name segment).

### Suggestion
- `type="email"` on an input that now intentionally accepts non-email syntax is inconsistent with the new logic - some browsers (Safari in particular) can still show a native "invalid email" affordance despite `noValidate` on the form. Switching to `type="text"` would be cleaner.
- No test for an empty display name before the bracket (`<noreply@openlinker.io>`) or for max-length combined with the display-name form - both called out in the plan as worth adding, neither landed in the test file.
- The regex requires exactly one space before `<` (e.g. `OpenLinker<noreply@x.io>` fails); the error message could clarify this, or the value could be trimmed before validation.

### Nitpick
- The new error message string is ~105 chars past indentation, over the 100-char Prettier `printWidth` - probably auto-wrapped fine, worth a quick lint check to confirm.
- No i18n for the new string, but that's consistent with existing convention in this file/feature, not a regression.

### What's fine as-is
- The two-regex-plus-helper approach (no RFC 5322 parsing library) is the right level of rigor for a cosmetic client-side check, matching this repo's MVP-appropriate philosophy - and the PR itself justifies this well in "Alternatives Considered".
- Scope is correctly limited to the frontend, no need to touch DTO/core/migrations.
- 102 lines of tests for a 12-line function isn't excessive - 7 meaningful cases, one assertion each.
- The plan doc follows the team's convention (single branch/PR with the plan included) and matches what shipped, aside from the test gap noted above.

**Recommendation:** fine to merge after adding a helper hint in the UI (Important #1) and deciding what to do about the regex gap (Important #2 - either a regression test documenting the accepted behavior, or a small regex fix). Everything else is a nice-to-have.

File: apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts [hunk 1]
```diff
@@ -19,7 +19,28 @@ export const MAX_FROM_ADDRESS_LENGTH = 320;
...
+// `(?!.*\.\.)` lookahead rejects a consecutive `..` anywhere (e.g.
+// `test..test@test.pl`), and the local-part/domain each require a non-dot
+// first character so neither segment can start with a stray `.`.
+const EMAIL_PATTERN = /^(?!.*\.\.)[^\s@<>.][^\s@<>]*@[^\s@<>.][^\s@<>]*\.[^\s@<>]+$/;
+// Matches `Display Name <email@domain.com>` — nodemailer parses this form natively
+// into the `From:` header, and the backend DTO has no `@IsEmail()` gate, so this
+// client-side check only needs to confirm the bracketed part looks like an email.
+// The name segment excludes `<`/`>` so a second bracketed address (e.g.
+// `A <b@c.com> <d@e.com>`) fails the match instead of silently picking the last one.
+const NAME_AND_EMAIL_PATTERN = /^[^<>]+\s<([^\s@<>]+@[^\s@<>]+\.[^\s@<>]+)>$/;
```
Comment (piotrswierzy) [inline apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts]: The name segment `[^<>]+` excludes `<`/`>` but **not** `\r`/`\n`, so a CRLF-carrying value validates as a "display name" address:

```
isValidFromAddress("Foo\r\nBcc: attacker@evil.com <a@b.com>") // => true
```

That is the classic email-header-injection shape (`From:` smuggling a `Bcc:`/extra header). The bare-email path is already CR/LF-safe because `\s` excludes them; the name segment should be too. Suggest `[^<>\r\n\t]+` (or reject any char < 0x20) and add a CRLF-in-name rejection test. IMPORTANT (defense-in-depth: `<input type=text>` strips newlines on entry and nodemailer encodes headers, but this pure validator should not rely on either).

**Review by piotrswierzy — COMMENTED** [review]
## Tech Lead Review — PR #1761 (`fix(web/mailer-settings): accept display-name From address`)

Thanks for the thorough plan doc and the well-targeted test file — the intent (relax the over-strict FE regex to match what the backend already accepts) is right, the DRY "capture-the-inner-email + one `EMAIL_PATTERN`" helper is a clean approach, and the tightened `EMAIL_PATTERN` (rejecting `..`, stray `<`/`>`, leading-dot segments) is a genuine improvement over the prior regex. The two-bracket / stray-bracket rejection tests are a nice touch.

One **IMPORTANT** security finding plus context on where the real validation boundary sits.

### IMPORTANT

**1. `NAME_AND_EMAIL_PATTERN` display-name segment permits CR/LF → header-injection-shaped value passes validation** (`mailer-settings-form.schema.ts:35`)

The name segment is `[^<>]+`, which excludes `<`/`>` but **not** `\r`/`\n` (unlike the bare-email path, where `\s` already excludes CR/LF). Verified:

```
isValidFromAddress("Foo\r\nBcc: attacker@evil.com <a@b.com>")  // => true
isValidFromAddress("Foo\nBcc: x@evil.com <a@b.com>")           // => true
```

A From value carrying an embedded CRLF is exactly the email-header-injection shape (`From:` → smuggled `Bcc:`/second header). The PR's own comment says this check exists to "confirm the bracketed part looks like an email" — so accepting a name segment with control chars contradicts the stated intent. Cheap defense-in-depth fix: exclude control chars from the name segment, e.g. `/^[^<>\r\n\t]+\s<(...)>$/` (or strip/reject any char < 0x20), and add a CRLF-in-name test alongside the existing malformed-input cases.

Mitigating factors (why IMPORTANT, not BLOCKING): the dialog uses `<input type="text">` which strips newlines on UI entry, and nodemailer encodes address headers downstream — but neither is a control this validator should rely on, and the function is a pure, reusable validator.

### Context / recommended follow-up (not blocking this FE-only PR)

**Backend is the actual trust boundary and currently validates nothing here.** `UpdateMailerSettingsDto.fromAddress` is `@IsOptional() @IsString()` only — no `@IsEmail()`, no CRLF guard — and the value flows unchanged into `db-backed-mailer.adapter.ts` → `SmtpMailerAdapter.sendMail({ from })` → nodemailer. Per `docs/frontend-architecture.md` (FE must not "duplicate backend validation ... as a source of truth"; "server-side validation remains the source of truth"), FE validation here is convenience-only and is trivially bypassed by a direct `PUT /mailer-settings`. The genuine header-injection surface therefore lives at the DTO, not this schema. This predates the PR and is out of its declared FE-only scope, but since the PR's justification is "the backend already accepts this," a follow-up adding a server-side CR/LF reject (e.g. a `@Matches` guard forbidding control chars) on `fromAddress` would close the real gap. Worth a tracking issue.

### Nits / SUGGESTION (non-blocking)

- `type="email"` → `type="text"` on the input is correct and necessary — a native `email` input would itself reject the display-name form. Good catch; no change needed.
- Consider a test asserting a control-char/CRLF name is rejected once the fix above lands, to lock the behavior in.

### Verdict

Functionally correct, good tests, matches backend behavior. Approving in principle but posting as **COMMENT** on the strength of the CRLF-in-display-name finding — please tighten the name segment (and ideally file the BE follow-up) before merge.

File: apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts [hunk 2]
```diff
@@ -19,7 +19,31 @@ export const MAX_FROM_ADDRESS_LENGTH = 320;
...
+const EMAIL_PATTERN = /^(?!.*\.\.)[^\s@<>.][^\s@<>]*@[^\s@<>.][^\s@<>]*\.[^\s@<>]+$/;
+// Matches `Display Name <email@domain.com>` — nodemailer parses this form natively
+// into the `From:` header, and the backend DTO has no `@IsEmail()` gate, so this
+// client-side check only needs to confirm the bracketed part looks like an email.
+// The name segment excludes `<`/`>` so a second bracketed address (e.g.
+// `A <b@c.com> <d@e.com>`) fails the match instead of silently picking the last one.
+// It also excludes CR/LF: unlike the bare-email path (where `\s` already blocks
+// them), `[^<>]` alone would let a CRLF-carrying name through — the classic
+// email-header-injection shape (`From:` value smuggling a second header).
+const NAME_AND_EMAIL_PATTERN = /^[^<>\r\n]+\s<([^\s@<>]+@[^\s@<>]+\.[^\s@<>]+)>$/;
```
Comment (piotrswierzy) [inline apps/web/src/features/mailer-settings/components/mailer-settings-form.schema.ts:38]: Resolved — the CRLF header-injection finding from my earlier review is closed. The name segment is now `[^<>\r\n]+`, so `"Foo\r\nBcc: attacker@evil.com <a@b.com>"` and the bare-LF variant both fail the match (`\s` already blocked them on the bare-email path). Two regression tests lock it (`embedded CRLF` and `embedded bare LF`), and the full schema spec passes (12/12 locally). No ReDoS in the pattern (single quantifiers, literal `.` anchors, no nested groups). Nice touch also hardening `EMAIL_PATTERN` against swallowed brackets and consecutive dots.

Non-blocking, and out of this FE PR's scope (as noted before): the real trust boundary is `UpdateMailerSettingsDto.fromAddress`, still only `@IsOptional() @IsString()` server-side — worth a follow-up issue to mirror this validation (or a header-safe check) at the DTO so the guarantee doesn't depend on the SPA.

**Review by piotrswierzy — APPROVED** [review]
## Re-review — CRLF header-injection finding RESOLVED ✅

Re-reviewed at head `cb4b1a7` (verified `git rev-parse FETCH_HEAD` == PR head SHA before trusting contents).

**My prior IMPORTANT finding is resolved.** Commit `8d259f9` (`fix: reject CRLF in the display-name segment`) narrows `NAME_AND_EMAIL_PATTERN`'s name segment from `[^<>]+` to `[^<>\r\n]+`. Walking the PoC: `"Foo\r\nBcc: attacker@evil.com <a@b.com>"` — the name segment can no longer span the `\r\n`, so there's no decomposition where `^[^<>\r\n]+\s<…>$` matches, and `EMAIL_PATTERN` already rejects it via `\s`. Both the CRLF and bare-LF shapes now return `false`. **Two regression tests lock the fix** (`should reject a display-name address carrying an embedded CRLF` and `… embedded bare LF`); ran the spec locally — **12/12 pass**.

Fresh pass on the rest:
- Bare-email path (`noreply@openlinker.io`) still accepted; display-name + valid inner email accepted — covered by tests.
- Extra hardening in the same wave (`EMAIL_PATTERN` now excludes `<`/`>` and rejects consecutive/leading dots) is correct and tested.
- No ReDoS — single quantifiers, disjoint char classes, literal `.`/`@` anchors, no nested groups.
- FE-only change; backend remains the source of truth (no BE files touched). No `any`, types clean.

One non-blocking note (unchanged from before, out of scope here): the actual trust boundary `UpdateMailerSettingsDto.fromAddress` is still only `@IsOptional() @IsString()` — worth a tracking issue to mirror this guard server-side.

Approving.
